Add buffers argument to comm open/close

vidartf opened the pull request:
Adds an argument `buffers` to `services.kernels.comm.Comm.open()`/`close()`. It is passed through to the `send_shell_message` similarly to what is done in `Comm.send()`.

Fixes #2730.

stonebig commented:
https://github.com/jupyter/notebook/issues/2730 ?

takluyver commented:
Thanks. Should `CommManager.prototype.new_comm` also gain a `buffers` parameter and pass it through to `comm.open()`?

takluyver commented:
@minrk @gnestor I've marked this as 5.1 since it's a fairly simple change that makes sense to me, and seems unlikely to have adverse effects.

vidartf commented:
@takluyver I added the buffers argument to `new_comm` as well. I added it at the end to retain compatibility with old signatures, even if the grouping of arguments then gets a little different.

takluyver commented:
Thanks, I think that's the right thing to do.

gnestor commented:
Thanks @vidartf 1
